### PR TITLE
convert_bypass_host_cache: add new case

### PR DESCRIPTION
**Pull request by lolyu**
check if `cache_mode` is `none`, `O_DIRECT` is on for system call `open`
or `openat`(RHEL8).

Signed-off-by: lolyu <lolyu@redhat.com>

### CODE REVIEW

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (47.90 s)
```

**Comment by lolyu** [comment]
depends on https://github.com/avocado-framework/avocado-vt/pull/2298

**Comment by lolyu** [comment]
@hellohellenmao @PaulYuuu Please help to review, thanks.

**Comment by PaulYuuu** [comment]
@zhenyzha Help to review ppc part, thanks.

**Comment by lolyu** [comment]
```
 (1/1) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.8.0.1.x86_64.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (48.39 s)
```

**Comment by hellohellenmao** [comment]
(1/1) Host_RHEL.m7.u8.luks.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: ERROR: 'str' object has no attribute '__module__' 

@lolyu Take a look, pls.

File: qemu/tests/image_convert_bypass_host_cache.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            msg = "%s is%spresented." % (flag, " " if negative else " not ")
+            msg += ", please refer to %s for more details" % strace_output_file
+            if any(flag in line for line in lines) is negative:
+                test.fail(msg)
+
+    root_dir = data_dir.get_data_dir()
+    strace_event = params["strace_event"].split()
+    image = params["images"]
+    image_params = params.object_params(image)
+    root_dir = data_dir.get_data_dir()
```
Comment (hellohellenmao) [inline qemu/tests/image_convert_bypass_host_cache.py]: Duplicated with line 26th.

**Comment by hellohellenmao** [comment]
Passed, and the log is okay to me. 
(1/1) Host_RHEL.m8.u1.luks.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (29.11 s)

**Comment by lolyu** [comment]
@YongxueHong please take a review.

**Comment by zhenyzha** [comment]
repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: CANCEL: Command 'strace' could not be found in any of the PATH dirs: ['/usr/sbin', '/usr/local/sbin', '/usr/local/bin', '/home/kar/workspace/bin', '/usr/bin', '/sbin', '/root/bin', '/usr/libexec', '/bin']

strace.rpm is not the default installation for PPC

**Comment by lolyu** [comment]
> repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: CANCEL: Command 'strace' could not be found in any of the PATH dirs: ['/usr/sbin', '/usr/local/sbin', '/usr/local/bin', '/home/kar/workspace/bin', '/usr/bin', '/sbin', '/root/bin', '/usr/libexec', '/bin']
> 
> strace.rpm is not the default installation for PPC

Yes, will add it to the optional-install rpms list in kar.

**Comment by zhenyzha** [comment]
(01/50) repeat1.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.39 s)
 (02/50) repeat2.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (03/50) repeat3.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.44 s)
 (04/50) repeat4.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.40 s)
 (05/50) repeat5.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (06/50) repeat6.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (07/50) repeat7.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (08/50) repeat8.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.39 s)
 (09/50) repeat9.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.36 s)
 (10/50) repeat10.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (11/50) repeat11.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (12/50) repeat12.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.36 s)
 (13/50) repeat13.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (14/50) repeat14.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.36 s)
 (15/50) repeat15.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (16/50) repeat16.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.36 s)
 (17/50) repeat17.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.41 s)
 (18/50) repeat18.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.52 s)
 (19/50) repeat19.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.52 s)
 (20/50) repeat20.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (21/50) repeat21.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (22/50) repeat22.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.42 s)
 (23/50) repeat23.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (24/50) repeat24.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (25/50) repeat25.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.39 s)
 (26/50) repeat26.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.49 s)
 (27/50) repeat27.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (28/50) repeat28.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (29/50) repeat29.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (30/50) repeat30.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (31/50) repeat31.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.40 s)
 (32/50) repeat32.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.36 s)
 (33/50) repeat33.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.50 s)
 (34/50) repeat34.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.39 s)
 (35/50) repeat35.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (36/50) repeat36.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (37/50) repeat37.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (38/50) repeat38.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (39/50) repeat39.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.40 s)
 (40/50) repeat40.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (41/50) repeat41.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.41 s)
 (42/50) repeat42.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.41 s)
 (43/50) repeat43.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.50 s)
 (44/50) repeat44.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (45/50) repeat45.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (46/50) repeat46.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (47/50) repeat47.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.36 s)
 (48/50) repeat48.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.37 s)
 (49/50) repeat49.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.38 s)
 (50/50) repeat50.Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.1.0.ppc64le.io-github-autotest-qemu.image_convert_bypass_host_cache.luks_to_luks: PASS (2.44 s)

LGTM.Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

**Comment by lolyu** [comment]
thanks all.

**Pull request merged**